.github/workflows/ci.yaml

```diff
@@ -1,54 +1,126 @@
+        with:
+          go-version: ^1.16
+      - uses: actions/setup-node@v2
+        with:
+          node-version: "15"
+      - uses: actions/cache@v2
+        id: go-cache
+        with:
+          path: ~/go/pkg/mod
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            ${{ runner.os }}-go-
+      - uses: actions/cache@v2
+        id: npm-cache
+        with:
+          path: ~/.npm
+          key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-node-
+      - name: install-mockery
+        run: go get github.com/vektra/mockery/v2
+      - name: test
+        run: make test
+      - name: static analysis
+        run: make vet-check
+      - name: coding styles
+        run: make fmt-check
+      - name: build
+        run: make -j4 cross-compiled
+      - name: compress
+        run: |
+          set -x
+          for FILE in build/*; do
+            gzip $FILE
+          done
+      - uses: actions/upload-artifact@v2
+        with:
+          name: trento-binaries
+          path: build
+      - uses: actions/upload-artifact@v2
+        with:
+          name: trento-amd64-binary
+          path: build/trento-amd64.gz
+      - uses: actions/upload-artifact@v2
+        with:
+          name: trento-examples-rules
+          path: examples
+
+  consul-cleanup-monitoring:
+    runs-on: [ self-hosted, vmmonitoring ]
+    needs: build
+    if: (github.event_name == 'push' && github.ref == 'refs/heads/main') || github.event_name == 'workflow_dispatch'
+    steps:
+      - uses: actions/checkout@v2
+      - id: cleanup
+        uses: ./.github/actions/consul_cleanup
+
+  deploy-hana01:
+    runs-on: [ self-hosted, vmhana01 ]
+    needs: consul-cleanup-monitoring
+    if: (github.event_name == 'push' && github.ref == 'refs/heads/main') || github.event_name == 'workflow_dispatch'
+    steps:
+      - uses: actions/checkout@v2
+      - id: cleanup
+        uses: ./.github/actions/consul_cleanup
+      - uses: actions/download-artifact@v2
+        with:
+          name: trento-amd64-binary
+      - uses: actions/download-artifact@v2
+        with:
+          name: trento-examples-rules
+      - id: deploy
+        uses: ./.github/actions/deploy
+        with:
+          role: agent
+
+  deploy-hana02:
+    runs-on: [ self-hosted, vmhana02 ]
+    needs: consul-cleanup-monitoring
+    if: (github.event_name == 'push' && github.ref == 'refs/heads/main') || github.event_name == 'workflow_dispatch'
+    steps:
+      - uses: actions/checkout@v2
+      - id: cleanup
+        uses: ./.github/actions/consul_cleanup
+      - uses: actions/download-artifact@v2
+        with:
+          name: trento-amd64-binary
+      - uses: actions/download-artifact@v2
+        with:
+          name: trento-examples-rules
+      - id: deploy
+        uses: ./.github/actions/deploy
+        with:
+          role: agent
```

Comment:
We weren't able to target two nodes with the same job since the `runs-on` directive expects all the tags to be present.
I will investigate more on the topic and get back if I find something!

Reply:
Also if the `runs-on` tags filter results in multiple nodes, just one action is executed in a round robin fashion, so just one runner will be actually running the deploy job.